addons/knowledge/static/src/js/knowledge_controller.js

```diff
@@ -106,6 +107,29 @@ const KnowledgeArticleFormController = FormController.extend({
         await this._rename(id, event.currentTarget.value);
     },
 
+    /**
+     * When the user clicks on the name of the article, checks if the article
+     * name hasn't been set yet. If it hasn't, it will look for a title in the
+     * body of the article and set it as the name of the article.
+     * @param {Event} event
+     */
+    _onSetArticleName: async function (event) {
+        const name = event.currentTarget.value;
+        if (name === _t('New Article')) {
```

Comment:
Checking against a translated term is not super great.
Should use a more robust check if possible.

Reply:
No we discussed, it is not really a pure onchange. It's ok, even if it makes me sad to have an onchange-like that is not an onchange :) .